File: branches/Medusa/Source/Fantasy.Jobs/JobPropertyReader.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Fantasy.Jobs
{
    public class JobPropertyReader : ObjectWithSite , ITagValueProvider 
    {
        #region ITagValueProvider Members

        public char Prefix
        {
            get { return '$'; }
        }

        public string GetTagValue(string tag, IDictionary<string, object> context)
        {
            Job job = (Job)this.Site.GetService(typeof(IJob));
            return job.GetProperty(tag); 

        }

        public bool HasTag(string tag, IDictionary<string, object> context)
        {
            Job job = (Job)this.Site.GetService(typeof(IJob));
            return job != null ? job.HasProperty(tag) : false;
        }

        public bool IsEnabled(IDictionary<string, object> context)
        {
            return this.Site != null && this.Site.GetService(typeof(IJob)) != null;
        }

        #endregion
    }
}

File: branches/Stheno/Fantasy.Jobs/src/fantasy/jobs/JobPropertyReader.java
package fantasy.jobs;

import fantasy.*;

public class JobPropertyReader extends ObjectWithSite implements ITagValueProvider
{


	public final char getPrefix()
	{
		return '$';
	}

	public final String GetTagValue(String tag, java.util.Map<String, Object> context) throws Exception
	{
		IJob job = this.getSite().getService(IJob.class);
		return job.GetProperty(tag);

	}

	public final boolean HasTag(String tag, java.util.Map<String, Object> context) throws Exception
	{
		IJob job = this.getSite().getService(IJob.class);
		return job != null ? job.HasProperty(tag) : false;
	}

	public final boolean IsEnabled(java.util.Map<String, Object> context) throws Exception
	{
		return this.getSite() != null && this.getSite().getService(IJob.class) != null;
	}

}